### PR TITLE
refactor: Introduce - replace custom mock with gomock

### PR DESCRIPTION
**Pull request by soluchok**
* gomock instead of  custom mock (introduce protocol)

Part of https://github.com/hyperledger/aries-framework-go/issues/269

Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -130,7 +130,8 @@ build-mockgen:
 .PHONY: mocks
 mocks: build-mockgen
 	$(call create_mock,pkg/client/introduce,Provider)
-	$(call create_mock,pkg/didcomm/protocol/introduce,InvitationEnvelope)
+	$(call create_mock,pkg/didcomm/protocol/introduce,Provider;InvitationEnvelope)
```
Comment (fqutishat) [inline Makefile]: we need to find way to clean mock file

Comment (fqutishat) [reply Makefile]: currently we have mock folder in each sub package

**Comment by rolsonquadras** [comment]
@soluchok a suggestion - Can we update the commit title to better align with the changes ie prefix with feature/protocol that we are changing ? eg. "refactor: Introduce - replace custom mock with gomock'. We use custom mocks in many places and the title without prefix might infer that we are replacing all of custom mocks with gomock.

File: Makefile (hunk 1 above)
Comment (soluchok) [reply Makefile]: sorry not sure I understand, I did not commit mock files, we only generate it on CI and local development (to run the tests). My point is to not commit mocks in the repository.

File: pkg/didcomm/protocol/introduce/service.go [hunk 2]
```diff
@@ -85,19 +85,20 @@ type Service struct {
...
+	wg          sync.WaitGroup
 	stop        chan struct{}
 	closedMutex sync.Mutex
 	closed      bool
 }
 
-// provider contains dependencies for the DID exchange protocol and is typically created by using aries.Context()
-type provider interface {
+// Provider contains dependencies for the DID exchange protocol and is typically created by using aries.Context()
+type Provider interface {
```
Comment (rolsonquadras) [inline pkg/didcomm/protocol/introduce/service.go]: Does this need to be exposed for gomocks ? just trying to understand the reason for this change.

Comment (soluchok) [reply pkg/didcomm/protocol/introduce/service.go]: Yep

File: Makefile (hunk 1 above)
Comment (troyronda) [reply Makefile]: @soluchok make clean should ideally remove all generated and build artifacts.

Comment (soluchok) [reply Makefile]: How about to replace it with `gomock` ? Or another name that is not so generic

Comment (soluchok) [reply Makefile]: I thought all mocks are located here `aries-framework-go/pkg/internal/mock`

Comment (soluchok) [reply Makefile]: @fqutishat @troyronda @rolsonquadras ^

Comment (soluchok) [reply Makefile]: Done!

File: Makefile [hunk 3]
```diff
@@ -130,10 +130,15 @@ build-mockgen:
...
+	$(call create_mock,pkg/didcomm/dispatcher,Outbound)
 	$(call create_mock,pkg/storage,Provider;Store)
 	$(call create_mock,pkg/didcomm/common/service,DIDComm)
 
+.PHONY: clean-mocks
+clean-mocks:
```
Comment (fqutishat) [inline Makefile]: Add clean-mocks to clean target

File: Makefile (hunk 1 above)
Comment (fqutishat) [reply Makefile]: looks good just add clean-mocks to clean target

File: Makefile (hunk 3 above)
Comment (soluchok) [reply Makefile]: there is an issue, looking for cause

Comment (soluchok) [reply Makefile]: Done!

**Pull request merged**